txmongo/_gridfs/__init__.py

```diff
@@ -177,6 +177,19 @@ def ok(cursor):
         return self.__files.find(query, filter=filter.sort(myorder), limit=1, skip=skip)\
             .addCallback(ok)
 
+    @defer.inlineCallbacks
+    def count(self, filename=None):
+        """Count the number of versions of a given file.
+        Returns an integer number of versions of the file in GridFS whose filename matches
+        `filename`, or raises NoFile if the file doesn't exist.
+        :Parameters:
+          - `filename`: ``"filename"`` of the file to get version count of
+        """
+        cursor = yield self.__files.find({"filename": filename})
+        if cursor:
+            defer.returnValue(len(cursor))
+        raise NoFile("no versions for filename %r" % (filename))
```

Comment:
That's strange for a function named "count" to raise an exception if there is no files. Intuition suggest that it should simply return zero. This way it will be more like same-named `Collection.count()`.